react/src/base/inputs/SprkSelect/SprkSelect.js

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import uniqueId from 'lodash/uniqueId';
+
+class SprkSelect extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      choiceItems: props.choices.map((item) => ({ id: uniqueId(), ...item })),
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.selectRef = React.createRef();
+  }
+
+  componentDidMount() {
+    /*
+     * Check if Huge Select has a value when it first mounts.
+     * Set presence of value in state.
+     */
+    const { variant } = this.props;
+    if (variant === 'huge') {
+      if (this.selectRef.current.value !== '') {
+        this.setState({
+          selectHugeHasValue: true,
+        });
+      }
+    }
+  }
+
+  /*
+   * Runs optional onChange prop.
+   * Updates state if huge selects have
+   * a value.
+   */
+  handleChange(e, variant, onChangeFunc) {
+    const isHugeSelect = variant === 'huge';
+
+    if (isHugeSelect) {
+      this.setState({
+        selectHugeHasValue: e.target.value !== '',
+      });
+    }
+    if (onChangeFunc) onChangeFunc(e);
+  }
+
+  render() {
+    const {
+      additionalClasses,
+      analyticsString,
+      isDisabled,
+      idString,
+      isValid,
+      id,
+      variant,
+      hasBlankFirstOption,
+      onChange,
+      ariaDescribedBy,
+      choices,
+      ...other
+    } = this.props;
+    const { choiceItems, selectHugeHasValue } = this.state;
+
+    return (
+      <select
+        className={classNames('sprk-b-Select', additionalClasses, {
+          'sprk-b-Input--has-floating-label': selectHugeHasValue,
+          'sprk-b-Select--error': !isValid,
+        })}
+        id={id}
+        disabled={isDisabled}
+        aria-describedby={ariaDescribedBy}
+        onChange={(e) => {
+          this.handleChange(e, variant, onChange);
+        }}
+        data-analytics={analyticsString}
+        data-id={idString}
+        ref={this.selectRef}
```

Comment:
Also, we'll need to update this with the multiple refs capability. Same for SprkInput etc.